### PR TITLE
[FIX] account: cash basis taxes: set move lines as tax_exigible=False only when some other line has a payable or receivable account in the move.

### PR DESCRIPTION
**Pull request by oco-odoo**
Before that, making a misc operation with cash basis taxes always set the move lines as tax_exigible=False (or at least, it was supposed to, see below). This was wrong, as if the move did not contain any payable/receivable account, it would result in no cash basis move being ever created, and so the entry would never affect the tax report.

Additionnally, the tax line was wrongly managed for cash basis taxes on misc entries, as vals.get('tax_ids') returned an ORM command, so a non-False value, even when the tax_ids field was empty. this caused the condition to execute and to reassign taxes to an empty list, which then caused tax_exigible not to be set to False, even when it should.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -126,17 +126,34 @@ def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu
...
+
         return res
 
+    @api.multi
+    def update_lines_tax_exigibility(self):
+        if all(account.user_type_id.type not in {'payable', 'receivable'} for account in self.mapped('line_ids.account_id')):
+            self.line_ids.write({'tax_exigible': True})
+        else:
+            tax_lines_caba = self.line_ids.filtered(lambda x: x.tax_line_id.tax_exigibility == 'on_payment')
+            base_lines_caba = self.line_ids.filtered(lambda x: any(tax.tax_exigibility == 'on_payment' for tax in x.tax_ids) )
```
Comment (smetl) [inline addons/account/models/account_move.py:152]: `x.tax_ids.flatten_taxes_hierarchy()`

File: addons/account/models/account_move.py [hunk 2]
```diff
@@ -126,17 +126,34 @@ def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu
...
             self.assert_balanced()
+            self.update_lines_tax_exigibility()
         else:
             res = super(AccountMove, self).write(vals)
+
         return res
 
+    @api.multi
+    def update_lines_tax_exigibility(self):
+        if all(account.user_type_id.type not in {'payable', 'receivable'} for account in self.mapped('line_ids.account_id')):
```
Comment (smetl) [inline addons/account/models/account_move.py:148]:
```
account_types = set(self.mapped('line_ids.account_id.user_type_id.type'))
if all(acc_type not in account_types for acc_type in ('payable', 'receivable')):
```?

**Review by smetl — CHANGES_REQUESTED** [review]

File: addons/account/models/account_move.py (hunk 1 above)
Comment (oco-odoo) [reply addons/account/models/account_move.py:152]: @smetl in v11? :thinking:

Comment (smetl) [reply addons/account/models/account_move.py:152]: I don't remember if you need to expand the group of taxes...

Comment (oco-odoo) [reply addons/account/models/account_move.py]: I didn't file any pre-made stuff for that in v11, so I changed the code a little bit to handle groups of taxes ;)

**Review by smetl — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**